### PR TITLE
feat(abstract-global-wallet-evm): add abstract global wallet connector

### PR DESCRIPTION
**Pull request by coffeexcoin**
This PR adds a custom connector for Abstract's Global Wallet.

This wallet is an ERC-4337 compatible wallet based (contracts at https://github.com/Abstract-Foundation/clave-contracts)

The wallet is managed by an embedded wallet signer secured by Privy's social login infrastructure. Signing is accomplished via a popup provided by the `@privy-io/cross-app-connect` package, which is a dependency of this

### CODE REVIEW

File: packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  override supportedChains: Chain[] = ["EVM", "ETH"];
+
+  override connectedChain: Chain = "EVM";
+
+  override findProvider(): IEthereum | undefined {
+    let chain = this.getActiveChain();
+    if (!chain) {
+      chain = abstractTestnet; // TODO: add mainnet
```
Comment (yonigoldberg) [inline packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts:72]: we should add mainnet as part of the release.

File: packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
+import { type EthereumWalletConnectorOpts } from '@dynamic-labs/ethereum-core';
+import { EthereumInjectedConnector, type IEthereum } from '@dynamic-labs/ethereum';
+import { toPrivyWalletProvider } from '@privy-io/cross-app-connect'
+import { transformEIP1193Provider } from '@abstract-foundation/agw-client';
+import { abstractTestnet } from 'viem/chains';
+import { DynamicError } from '@dynamic-labs/utils';
+import { type Chain, logger } from '@dynamic-labs/wallet-connector-core';
+import { findWalletBookWallet } from '@dynamic-labs/wallet-book';
+import { toHex } from 'viem';
+const AGW_APP_ID = 'cm04asygd041fmry9zmcyn5o5';
```
Comment (yonigoldberg) [inline packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts:10]: Is it ok to hardcode this?

File: packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  override findProvider(): IEthereum | undefined {
+    let chain = this.getActiveChain();
+    if (!chain) {
+      chain = abstractTestnet; // TODO: add mainnet
+    } 
+    
+    const privyProvider = toPrivyWalletProvider({
+      providerAppId: AGW_APP_ID,
+      chains: [abstractTestnet] // TODO: add mainnet
```
Comment (yonigoldberg) [inline packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts:77]: same

File: packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+   */
+  override supportsNetworkSwitching(): boolean {
+    return false;
+  }
+
+  override isInstalledOnBrowser(): boolean {
+    return true;
+  }
+
+  override async init(): Promise<void> {
```
Comment (yonigoldberg) [inline packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts:48]: do we need to add a check that abstract chain is enabled? (this come from evmNetworks in the props)

**Review by yonigoldberg — COMMENTED** [review]
We need to cherry-pick also to the V3 branch (ME is on it right now)

File: packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts (hunk 2 above)
Comment (coffeexcoin) [reply packages/@dynamic-labs-connectors/abstract-global-wallet-evm/src/AbstractEvmWalletConnector.ts:10]: Yes - this is a public app id used to identify us in the cross app connection process.

**Review by yonigoldberg — APPROVED** [review]
LGTM - please cherry-pick to the v3  branch as well so we can release both

**Pull request merged**